internal/ssh/ssh_test.go

```diff
@@ -189,47 +195,56 @@ func (s *sshSuite) TestSSHJump(c *qt.C) {
 }
 
 func (s *sshSuite) TestSSHJumpPermissionFail(c *qt.C) {
-	client, err := gossh.Dial("tcp", fmt.Sprintf(":%d", s.jumpServerPort), &gossh.ClientConfig{
-		HostKeyCallback: gossh.FixedHostKey(s.hostKey.PublicKey()),
-		Auth: []gossh.AuthMethod{
-			gossh.PublicKeys(s.privateKey),
+	tests := []struct {
+		name     string
+		user     string
+		destAddr string
+		errMsg   string
+	}{
+		{
+			name:     "alice not allowed on this model",
+			user:     "alice",
+			destAddr: "982b16d9-a945-4762-b684-fd4fd885aa11",
+			errMsg:   "ssh: rejected: connect failed (user doesn't have permission)",
+		},
+		{
+			name:     "bob not allowed on this model",
+			user:     "bob",
+			destAddr: s.allowedModelUUID,
+			errMsg:   "ssh: rejected: connect failed (user doesn't have permission)",
+		},
+		{
+			name:     "not existing user",
+			user:     "mark",
+			destAddr: s.allowedModelUUID,
+			errMsg:   "ssh: rejected: connect failed (user doesn't have permission)",
 		},
-		User: "alice",
-	})
-	c.Assert(err, qt.IsNil)
-	defer client.Close()
-
-	// send forward message
-	msg := ssh.ForwardMessage{
-		DestAddr: "982b16d9-a945-4762-b684-fd4fd885aa11",
-		//nolint:gosec
-		DestPort: uint32(s.destinationServerPort),
-		SrcAddr:  "localhost",
-		SrcPort:  0,
 	}
-	_, _, err = client.OpenChannel("direct-tcpip", gossh.Marshal(&msg))
-	c.Assert(err, qt.ErrorMatches, ".*user doesn't have permission.*")
 
-	client, err = gossh.Dial("tcp", fmt.Sprintf(":%d", s.jumpServerPort), &gossh.ClientConfig{
-		//nolint:gosec // this will be removed once we handle hostkeys
-		HostKeyCallback: gossh.InsecureIgnoreHostKey(),
-		Auth: []gossh.AuthMethod{
-			gossh.PublicKeys(s.privateKey),
-		},
-		User: "bob",
-	})
-	c.Assert(err, qt.IsNil)
-	defer client.Close()
-	// send forward message
-	msg = ssh.ForwardMessage{
-		DestAddr: s.allowedModelUUID,
-		//nolint:gosec
-		DestPort: uint32(s.destinationServerPort),
-		SrcAddr:  "localhost",
-		SrcPort:  0,
+	for _, test := range tests {
+		c.Run(test.name, func(c *qt.C) {
+			client, err := gossh.Dial("tcp", fmt.Sprintf(":%d", s.jumpServerPort), &gossh.ClientConfig{
+				HostKeyCallback: gossh.FixedHostKey(s.hostKey.PublicKey()),
+				Auth: []gossh.AuthMethod{
+					gossh.PublicKeys(s.privateKey),
+				},
+				User: test.user,
+			})
+			c.Assert(err, qt.IsNil)
+			defer client.Close()
+
+			// send forward message
+			msg := ssh.ForwardMessage{
+				DestAddr: test.destAddr,
+				//nolint:gosec
+				DestPort: uint32(s.destinationServerPort),
+				SrcAddr:  "localhost",
+				SrcPort:  0,
+			}
+			_, _, err = client.OpenChannel("direct-tcpip", gossh.Marshal(&msg))
```

Comment:
I'll check if there isn't a simpler way to do this

Reply:
Yeah, instead of doing all this manually, the client offers this by just doing client.Dial, I've tested the below and your test passes (also remove the struct),
```suggestion
			_, err = client.Dial("tcp", fmt.Sprintf("%s:%d", test.destAddr, s.destinationServerPort))
			c.Assert(err.Error(), qt.Equals, test.errMsg)
```

Reply:
thanks for this